polkadot-protocol/basics/data-encoding.md

```diff
@@ -1,4 +1,105 @@
 ---
 title: Data Encoding
-description: TODO
----
\ No newline at end of file
+description: SCALE codec enables fast, efficient data encoding in Substrate, ideal for resource-constrained environments like Wasm, supporting custom types and compact encoding.
+---
+
+# Data Encoding
+
+Polkadot SDK uses a lightweight and efficient encoding/decoding mechanism to optimize data transmission across the network. This mechanism, known as the **SCALE** codec, is used for serializing and deserializing data.
+
+The **SCALE** codec plays a vital role in enabling communication between the runtime and the outer node.
+
+The **SCALE** codec is designed for high-performance, copy-free data encoding and decoding in resource-constrained environments like the Polkadot SDK [WebAssembly runtime](add link to Runtime development page){target=\_blank}.
+
+It is not self-describing, meaning the decoding context must have full knowledge of the encoded data types. 
+
+Parity's front-end libraries utilize the [`parity-scale-codec`](https://github.com/paritytech/parity-scale-codec){target=\_blank} crate (a Rust implementation of the SCALE codec) to handle encoding and decoding for interactions between RPCs and the runtime.
```

Comment:
What is the meaning of `front-end` in this context? I'm unsure if this is accurate, as it mentions a Rust crate. I doubt Polkadot.js API, for example, uses the rust crate under the hood. It is probably a js implementation of scale